### DISCUSSION TITLE
Optimize the S6 Images work with PaaS hosts (fixing the error "s6-overlay-suexec: fatal: can only run as pid 1")

### DISCUSSION THREAD

**Discussion by jaydrogers** [serversideup/docker-php#191] [Ideas & Feature Requests]
# Problem
Some hosts expect to have `PID 1` and it causes this error to display:

```
s6-overlay-suexec: fatal: can only run as pid 1
```

### Known Hosts with this issue
- Fly.io
- Heroku
-  ECS Fargate

### References:
These are the related discussions we've seen to this issue:
- https://discord.com/channels/910287105714954251/910299240897605632/1095593392035082341
- https://github.com/serversideup/docker-php/discussions/87
- https://github.com/serversideup/docker-php/discussions/189

# Proposed solution
- [ ] Reach out to the team at S6 Overlay to see if they have any known workarounds (https://github.com/just-containers/s6-overlay)
- [ ] Reach out to the PaaS hosts and see if they support releasing `PID 1` to S6 overlay

**Comment by jaydrogers** [comment]
### Update: S6 Overlay maintainer has some thoughts
One of the maintainers of S6 Overlay commented on this here: https://github.com/just-containers/s6-overlay/issues/535#issuecomment-1597438481

> It sounds like ECS Fargate doesn't give you control over your process 1 by default. You should check with their documentation whether they are running their own init and if there's a way to bypass it.
>
> If it's unavoidable, then they're not providing you with a real container environment, but a functionality-reduced simile of it, and you'll be unable to run s6-overlay on that platform. Sorry.

**Comment by moisish** [comment]
For the ECS, I think i have found the issue. By disabling the init process, everything works as expected

```
InitProcessEnabled
Run an init process inside the container that forwards signals and reaps processes. This parameter maps to the --init option to [docker run](https://docs.docker.com/engine/reference/run/#security-configuration).
```

**Comment by mpskovvang** [comment]
I was struggling with the same error on Fly.io.

Switching the image variant from FPM-NGINX to Unit solved the issue with no further changes since Unit is not using the S6 Overlay.
